arcade/ticket/src/main.leo

```diff
@@ -0,0 +1,65 @@
+program puzzle_arcade_ticket_v002.aleo {
+
+  // game programs that can mint tickets
+  mapping registry: address => bool;
+
+  record PuzzleArcadeTicket {
+    owner: address,
+    amount: u64
+  }
+
+  async transition add_program_to_registry(program_address: address) -> (Future) {
+    assert_eq(self.caller, self.signer);
+    assert_eq(self.caller, aleo1pedhtu6akw9z68wedu8t3fgxfdh3ye2rypeqkx9cxjpr99chqvyqkjg7rw);
+    return register_program(program_address);
+  }
+
+  async function register_program(program_address: address) {
+    Mapping::set(registry, program_address, true);
```

Comment:
probably want to add a check whether the program already exists in the registry mapping